### ISSUE TITLE
Collision between pages that are named the same in 2 different domains and creating redirection problem since update

### ISSUE THREAD

**Issue by daveletourneau** [dmitryd/typo3-realurl#46]
I have two different sites (and domains) configured that each have their homepage named "accueil" (home in french). Everything was ok since i've upgraded to 2.x but now only one site can work at once, the other throwing a redirection error. I've manually applied the scripts mentioned in the upgrade procedure after upgrading and clearing the cache always produce the same result. It may be useful to mention that each root page of the sites are shortcuts (since I just saw an issue with them on the tracker).

I don't really know or master the guts of realURL but I saw that in the _tx_realurl_urlcache_ table that the rootpage seems to always be 1 even if the second site starts with 359. Here are 2 screenshots to best describe the problem...

![selection_072](https://cloud.githubusercontent.com/assets/6052392/13538784/7c0654fe-e21a-11e5-8e3b-ec082de7b6da.png)
![selection_073](https://cloud.githubusercontent.com/assets/6052392/13538785/7c06c0ec-e21a-11e5-8f59-0698cbcb6b9c.png)

**Comment by dmitryd** [comment]
Thanks for the detailed report!

It looks like root page id is not determined correctly. Could you check that you have proper domain records on pids 1 and 359?

**Comment by daveletourneau** [comment]
They are both there (and were working Ok before)

![selection_074](https://cloud.githubusercontent.com/assets/6052392/13539021/b2883ed8-e21b-11e5-9a0c-b47ce51abc67.png)
![selection_077](https://cloud.githubusercontent.com/assets/6052392/13539020/b287b12a-e21b-11e5-9e6c-48c5d72b6482.png)

**Comment by dmitryd** [comment]
Also, please, have a look at Admin Tools > Configuration. Under TYPO3_CONF_VARS > EXTCONF > realurl you should see your domains. Inside them you should see a `pagePath` entry. Both domains should have a correct `rootpage_id` value.

![config](https://cloud.githubusercontent.com/assets/306133/13539112/72f61738-e25f-11e5-9413-e563c5be5fab.png)

**Comment by daveletourneau** [comment]
I did not have the domains listed there but it made me think of something else and think I found the source of the problem. I used a custom configuration to enable realURL to process parameters from my own extension and this config file only defines the '_DEFAULT' branch. I enabled the automatic config and the domains were back in the EXTCONF > realurl node.

So, if I understand, all I have to do is to modify the _$GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['realurl']_ array to list the two domains. Makes sense but it's weird that it was working before the update. Anyways, sorry for the trouble and thank you for the excellent support!
